Недочёт в обёртке для DAWG библиотеки
=====================================
:date: 2013-05-25 03:37:32
:tags: ошибка, исправление, библиотека, python, обёртка, DAWG, cython
:category: text
:author: IMDagger
:yaru-link: http://imdagger.ya.ru/replies.xml?item_no=1801

    Отправил \ `pull-запрос <https://github.com/kmike/DAWG/pull/10>`__ с
исправлением небольшой неточности при открытии файлов в библиотеке DAWG
для Python. Сначала естественно написал и закоммитил тест, а потом уже
проверил свои мысли и исправления на нём. Заодно немного почитал про
Cython, т.к. не понял сходу почему он не принял мой bool (а оказалось,
что их два вида и нужно импортировать необходимый). Заодно и из курса по
C++ освежил в памяти, как проверять поток на ошибки, что есть флаги,
есть ещё удобные методы для проверки флагов: .fail(), .bad(), .good().
Вот и добавил, перед попыткой прочитать данные из потока при помощи
внутренней библиотеки, проверку на состояние с выбросом исключения.

